modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/dashboard/ProductPerfomance/businessRules/index.ts

```diff
@@ -114,52 +180,107 @@ export async function sixMonthRule(
 		sixMonthsAgoDate[0],
 		sixMonthsAgoDate[1]
 	);
-	const lastSixMonthsGoalsResult = getArrayOfGoals(
-		salesGoal,
-		sixMonthsGoalsArray
-	);
 
 	const policiesForSalesGoal = await getPoliciesForSalesGoal(
 		currentDateString[0],
 		currentDateString[1],
 		sixMonthsAgoDate[0],
 		sixMonthsAgoDate[1]
 	);
-	const lastSixMonthsSalesResult = getArrayOfSales(
-		policiesForSalesGoal,
-		sixMonthsSalesArray
+
+	const productsType = salesGoal.data.items.filter((item: SalesGoal) => {
+		if (
+			item.productExternalReferenceCode === productExternalReferenceCode
+		) {
+			return item;
+		}
+		else if (productExternalReferenceCode === 'All') {
+			return item;
+		}
+	});
+	const productsSalesType = policiesForSalesGoal.data.items.filter(
+		(item: SalesGoal) => {
+			if (
+				item.productExternalReferenceCode ===
+				productExternalReferenceCode
+			) {
+				return item;
+			}
+			else if (productExternalReferenceCode === 'All') {
+				return item;
+			}
+		}
+	);
+	const sixMonthsGoalsResultProducts = getArrayOfGoalsProducts(
+		productsType,
+		sixMonthsGoalsArray,
+		productExternalReferenceCode
+	);
+
+	const sixMonthsSalesResultProducts = getArrayOfSalesProducts(
+		productsSalesType,
+		sixMonthsSalesArray,
+		productExternalReferenceCode
 	);
 
-	return [lastSixMonthsGoalsResult, lastSixMonthsSalesResult];
+	return [sixMonthsGoalsResultProducts, sixMonthsSalesResultProducts];
 }
 
 export async function threeMonthRule(
 	currentDateString: string[],
 	threeMonthsAgoDate: string[],
 	threeMonthsGoalsArray: string[],
-	threeMonthsSalesArray: string[]
+	threeMonthsSalesArray: string[],
+	productExternalReferenceCode: string
 ) {
 	const salesGoal = await getSalesGoal(
 		currentDateString[0],
 		currentDateString[1],
 		threeMonthsAgoDate[0],
 		threeMonthsAgoDate[1]
 	);
-	const lastThreeMonthsGoalsResult = getArrayOfGoals(
-		salesGoal,
-		threeMonthsGoalsArray
-	);
 
 	const policiesForSalesGoal = await getPoliciesForSalesGoal(
 		currentDateString[0],
 		currentDateString[1],
 		threeMonthsAgoDate[0],
 		threeMonthsAgoDate[1]
 	);
-	const lastThreeMonthsSalesResult = getArrayOfSales(
-		policiesForSalesGoal,
-		threeMonthsSalesArray
+
+	const productsType = salesGoal.data.items.filter((item: SalesGoal) => {
+		if (
+			item.productExternalReferenceCode === productExternalReferenceCode
+		) {
+			return item;
+		}
+		else if (productExternalReferenceCode === 'All') {
+			return item;
+		}
+	});
+	const productsSalesType = policiesForSalesGoal.data.items.filter(
+		(item: SalesGoal) => {
+			if (
+				item.productExternalReferenceCode ===
+				productExternalReferenceCode
+			) {
+				return item;
+			}
+			else if (productExternalReferenceCode === 'All') {
+				return item;
+			}
+		}
+	);
```

Comment:
You should use a generic function for this like:
```suggestion
   function compareProducts (data) {
    data.filter((item: dataItem) => {
		if (
			item.productExternalReferenceCode === productExternalReferenceCode
		) {
			return item;
		}
		else if (productExternalReferenceCode === 'All') {
			return item;
		}
	});
	
```

Reply:
Done. 
Thx!!